packages/web/src/common/store/notifications/parseAndProcessNotifications.ts

```diff
@@ -0,0 +1,202 @@
+        collectionIdsToFetch.push(...notification.entityIds)
+      }
+      userIdsToFetch.push(notification.userId)
+    }
+    if (
+      notification.type === NotificationType.Repost ||
+      notification.type === NotificationType.Favorite ||
+      (notification.type === NotificationType.Milestone &&
+        'entityType' in notification)
+    ) {
+      if (notification.entityType === Entity.Track) {
+        trackIdsToFetch.push(notification.entityId)
+      } else if (
+        notification.entityType === Entity.Playlist ||
+        notification.entityType === Entity.Album
+      ) {
+        collectionIdsToFetch.push(notification.entityId)
+      } else if (notification.entityType === Entity.User) {
+        userIdsToFetch.push(notification.entityId)
+      }
+    }
+    if (
+      notification.type === NotificationType.Follow ||
+      notification.type === NotificationType.Repost ||
+      notification.type === NotificationType.Favorite
+    ) {
+      // @ts-ignore
+      notification.userIds = [...new Set(notification.userIds)]
+      userIdsToFetch.push(
+        ...notification.userIds.slice(0, USER_INITIAL_LOAD_COUNT)
+      )
+    }
+    if (notification.type === NotificationType.RemixCreate) {
+      trackIdsToFetch.push(
+        notification.parentTrackId,
+        notification.childTrackId
+      )
+      notification.entityType = Entity.Track
+      notification.entityIds = [
+        notification.parentTrackId,
+        notification.childTrackId
+      ]
+    }
+    if (notification.type === NotificationType.RemixCosign) {
+      trackIdsToFetch.push(notification.childTrackId)
+      userIdsToFetch.push(notification.parentTrackUserId)
+      notification.entityType = Entity.Track
+      notification.entityIds = [notification.childTrackId]
+      notification.userId = notification.parentTrackUserId
+    }
+    if (notification.type === NotificationType.TrendingTrack) {
+      trackIdsToFetch.push(notification.entityId)
+    }
+    if (
+      notification.type === NotificationType.TipSend ||
+      notification.type === NotificationType.TipReceive ||
+      notification.type === NotificationType.SupporterRankUp ||
+      notification.type === NotificationType.SupportingRankUp ||
+      notification.type === NotificationType.Reaction
+    ) {
+      userIdsToFetch.push(notification.entityId)
+    }
+    if (notification.type === NotificationType.TipReceive) {
+      reactionSignatureToFetch.push(notification.tipTxSignature)
+    }
+    if (notification.type === NotificationType.AddTrackToPlaylist) {
+      trackIdsToFetch.push(notification.trackId)
+      userIdsToFetch.push(notification.playlistOwnerId)
+      collectionIdsToFetch.push(notification.playlistId)
+    }
+    if (notification.type === NotificationType.SupporterDethroned) {
+      userIdsToFetch.push(notification.supportedUserId)
+      userIdsToFetch.push(notification.entityId)
+    }
+  })
```

Comment:
This is a little cumbersome to parse. Wondering if there's a better structure for the code?
Maybe even just some line separation and/or those section header comments we used in offline sagas.

Reply:
Alternatively, we could separate the checks into different handlers responsible for each category

Reply:
also totally agreed